Build with Nix

ttuegel opened the pull request.

dwightguth commented:
Jenkins: test this please

dwightguth commented:
I am merging this since it seems to fix the frequent test failures in master that we have had.

@andreistefanescu if you want to add back this test you will have to solve the segfault issue. I tried just changing it to --z3-executable but that apparently causes the test to fail. I think you should seriously consider whether you want to keep spending effort writing code that only works for a mode of verification that can cause an extremely simple test like this to fail over 50% of the time.

ttuegel committed:
nix: Import expressions from nixpkgs-kframework

ttuegel committed:
gitignore: .depend

ttuegel committed:
pom.xml: fix path to haskell-backend

ttuegel committed:
ktest.mak: allow overriding path to kx

ttuegel committed:
gitignore: remove .depend

ttuegel committed:
nix/k.nix: recursive gitignore

ttuegel committed:
Add test.nix

ttuegel committed:
nix/k.nix: use gitignoreSourcePure

gitignoreFilterRecursiveSource is terribly slow, and it doesn't do the right
thing anyway. gitignoreSourcePure will be fine as long as we are using a clean
Git checkout, which will always be the case for CI.

ttuegel committed:
nix/k.nix: check for prelude.kore

ttuegel committed:
nix/k.nix: ignore irrelevant haskell-backend sources

The only file from haskell-backend which is used in the frontend build is
`prelude.kore`.

ttuegel committed:
nix/k.nix: ignore irrelevant llvm-backend sources

ttuegel committed:
parseNonPgm: override KRUN

ttuegel committed:
ktest.mak: Override KRUN for Kore backends

ttuegel committed:
ktest.mak: Use user-specified path to krun, if any

ttuegel committed:
krun-deserialize: Override KX instead of KRUN

ttuegel committed:
Exclude regression tests from frontend build

ttuegel committed:
test.nix: Include only regression tests

ttuegel committed:
test.nix: Run patchShebangs

ttuegel committed:
test.nix: Allow user to specify single test on command line

ttuegel committed:
test.nix: Do not save test products

The test products are almost 1.5 GB in size.

ttuegel committed:
Add Test workflow
